app/routes.py

```diff
@@ -0,0 +1,286 @@
+
+        db.session.add(new_video)
+        db.session.commit()
+        return make_response({"id": new_video.id, "title": new_video.title, "total_inventory": new_video.total_inventory}, 201)
+
+# Wave 1 GET/DELETE/PUT
+@videos_bp.route("", methods=["GET"])
+def get_all_videos():
+    videos = Video.query.all()
+    response_body = [video.to_dict() for video in videos]
+    return jsonify(response_body), 200
+
+@videos_bp.route("/<video_id>", methods=["GET", "DELETE", "PUT"])
+def handle_video(video_id):
+    if video_id.isnumeric() != True:
+        return {"details" : "Invalid request"}, 400
+    video = Video.query.get(video_id)
+    
+    
+    if request.method == "GET":
+        if video is None:
+            return make_response({"message": f"Video {video_id} was not found"}, 404)
+
+        
+        return make_response(video.to_dict(), 200)
+
+    elif request.method == "DELETE":
+        if video is None:
+            return make_response({"message": f"Video {video_id} was not found"}, 404)
+
+        db.session.delete(video)
+        db.session.commit()
+        
+        return make_response({"id": int(video_id)}, 200)
+
+    if request.method == "PUT":
+        request_body = request.get_json()
+        if video is None:
+            return make_response({"message": f"Video {video_id} was not found"}, 404)
+        elif  "title" not in request_body or "release_date" not in request_body or "total_inventory" not in request_body:
+            return make_response({"details": "Invalid data"}, 400)
+        else:
+            response_body = request.get_json()
+        
+        video.title = response_body["title"]
+        video.release_date = response_body["release_date"]
+        video.total_inventory = response_body["total_inventory"]
+    
+        db.session.commit()
+        return make_response(video.to_dict(), 200)
+
+# WAVE 2
+
+@rentals_bp.route("/check-out", methods=["POST"])
+def create_rental():
+
+    request_body = request.get_json()
+
+    if "customer_id" not in request_body or "video_id" not in request_body:
+        abort(400, "Request body must include a customer_id and video_id")
+
+    try:
+        customer_id = int(request_body["customer_id"])
+        video_id = int(request_body["video_id"])
+    except ValueError:
+        return jsonify({"Error": "Customer ID and Video ID must be integers."}), 400
+
+    customer = Customer.query.get(customer_id)
+    video = Video.query.get(video_id)
+
+    if customer is None:
+        return jsonify({"message": f"Customer {customer_id} was not found"}), 404
+    elif video is None:
+        return jsonify({"message": f"Video {video_id} was not found"}), 404
+
+    if video.total_inventory == 0:
+        return jsonify({"message": f"Video {video_id} is out of stock"}), 404
+
+    if video.total_inventory - len(video.rentals) == 0:
+        abort(make_response({"message": "Could not perform checkout"}, 400))
```

Comment:
There is a small logical bug here.  The list of rentals associated with a video will include all of the rentals where that have a foreign key `video_id` that matches the id of this video.  The check-in logic sets the checked-out variable of the rental to False, but does not delete the rental.  This list will then contain all of the current and past rentals.  One way to fix this is to go through the list of rentals and count the ones that are active:
```suggestion
    active_rentals = 0
    for rental in video.rentals:
        if rental.checked_out:
            active_rentals += 1
            
    if video.total_inventory - active_rentals == 0:
        abort(make_response({"message": "Could not perform checkout"}, 400))
```